src/hotspot/share/runtime/deoptimization.cpp

```diff
@@ -1929,6 +1932,19 @@ JRT_ENTRY(void, Deoptimization::uncommon_trap_inner(JavaThread* current, jint tr
 
     JFR_ONLY(post_deoptimization_event(nm, trap_method(), trap_bci, trap_bc, reason, action);)
 
+    if(Log(deoptimization)::is_info()) {
+      LogMessage(deoptimization) lm;
+      auto tm = trap_method();
```

Comment:
Not sure use if C++11 type inference is a permitted use - I can't find it n the style guide.

Reply:
`LogMessage` ensures that messages are kept in order in the final output, but this is not needed if we're logging on one line as Jorn has suggested. I'm also removing the type inference.

Reply:
[not a review, just responding to @dholmes-ora question]

Search for "Type Deduction" in the style guide.  There aren't a lot of uses, and the guidance is to only use it when there's significant benefit, not just as a convenience to avoid writing the type.